bootstrap/pkg/kfapp/gcp/gcp.go

```diff
@@ -672,10 +672,37 @@ func (gcp *Gcp) deleteEndpoints(ctx context.Context) error {
 	services := servicemanagement.NewServicesService(servicemanagementService)
 	op, err := services.Delete(gcp.Spec.Hostname).Context(ctx).Do()
 	if err != nil {
-		return &kfapis.KfError{
-			Code:    int(kfapis.INTERNAL_ERROR),
-			Message: fmt.Sprintf("issuing endpoint deletion error: %v", err),
+		nextPage := ""
+		for {
+			list := services.List().ProducerProjectId(gcp.Spec.Project)
+			if nextPage != "" {
+				list = list.PageToken(nextPage)
+			}
+			listResp, err := list.Do()
+			if err != nil {
+				return &kfapis.KfError{
+					Code:    int(kfapis.INTERNAL_ERROR),
+					Message: fmt.Sprintf("listing managed services error: %v", err),
+				}
+			}
+			for _, s := range listResp.Services {
+				if s.ServiceName == gcp.Spec.Hostname {
+					return &kfapis.KfError{
+						Code:    int(kfapis.INTERNAL_ERROR),
+						Message: fmt.Sprintf("issuing endpoint deletion error: %v", err),
```

Comment:
this err is the error from list, should use the delete error?

Reply:
this is because deletion is failed and we could still find endpoint service in the list - this is an actual deletion error.

Reply:
I mean, the err here is from list.Do.
shouldn't we return the err from services.Delete?

Reply:
changed.